US003 - View Capability a role belongs to

OdhranHague opened the pull request:
As a Kainos employee

I want to be able to see what capability each role falls within
So that I can see what capability the role is within Kainos corporate structure

Acceptance Criteria

Update the job roles list page to show the capability name each role belongs to

- Updated getRoles query to include the "CapabilityName" field
- Updated "Role" class constructor to include capability name

OdhranHague committed:
model for role class

OdhranHague committed:
view roles

OdhranHague committed:
tweak to roles.html

OdhranHague committed:
install typescript added to workflow

OdhranHague committed:
Changed file name

OdhranHague committed:
capability field added

jkelso662 commented:
Would be good to always have a screenshot for UI PRs :)

jkelso662 commented on views/roles.html:
```diff
@@ -0,0 +1,51 @@
+                        <td>{{role.specification}}</td>
+                        <td>{{role.capabilityName}}</td>
+                        <td>{{role.responsibilities}}</td>
+                        <td><a href="role.sharepointLink">Sharepoint</a></td>
```
🐛

jkelso662 reviewed (changes requested):
Think I spotted a bug might need checked out. Also you have a duplicated Role.ts file. Sort these and maybe and a screenshot of your UI and can get this approved :)

OdhranHague committed:
Delete model/role.ts

git wouldnt let me change the file name from 'role.ts' -> 'Role.ts'
turned off the ignore case config setting but still not working

OdhranHague committed:
Merge branch 'main' into US001-view-job-roles

OdhranHague committed:
changed error message

OdhranHague committed:
changed linter config

OdhranHague committed:
changed linter

OdhranHague committed:
Delete model/role.ts

OdhranHague committed:
Merge US038

OdhranHague commented:
- UI for viewing capability
![Screenshot 2023-12-12 at 14 03 23](https://github.com/Criofann/JobForge-UI/assets/145450866/2a7e7950-a811-4a9e-9a42-ceb52d90a506)

I have been running into issues with that duplicated role.ts file. The "Role.ts" is the only one there locally and when i pull down "role.ts", it never appears. I have just manually deleted the file and committed it through Github

Joseph-Magee reviewed (approved):
looks good

lullabelle reviewed (approved):
changes approved

UserConor reviewed (approved):
Looks good, but will need the new UI implemented once it's complete
